src/responder/kcm/kcmsrv_cmd.c

```diff
@@ -543,7 +543,8 @@ static void kcm_recv(struct cli_ctx *cctx)
         DEBUG(SSSDBG_FATAL_FAILURE,
             "Failed to parse data (%d, %s), aborting client\n",
             ret, sss_strerror(ret));
-        goto fail;
+        talloc_free(cctx);
```

Comment:
In this function `talloc_free(cctx)` and `return` is used everywhere (only one case is special). Maybe it is worth to refactor this so `talloc_free(ctx)` will be part of `fail:` jump?

Reply:
I removed the fail label.